src/mumble/AudioInput.h

```diff
@@ -119,8 +119,18 @@ class Resynchronizer {
 	enum { S0, S1a, S1b, S2, S3, S4a, S4b, S5 } state = S0; ///< Queue fill control statemachine
 };
 
-class AudioInputRegistrar {
+struct EchoCancellationOption {
+	EchoCancellationOption(int id, QString description, QString explanation) : id(id),
+	                                                                           description(description),
+	                                                                           explanation(explanation) {};
+	int id;
+	QString description;
+	QString explanation;
+};
+
+class AudioInputRegistrar : public QObject {
 private:
+	Q_OBJECT
```

Comment:
Why do we have to turn the registrar into a `QObject`?

Reply:
Because I need to use the "tr" function for the description of each echoOption.

Reply:
I may be mistaken but from what I know `tr` is a static function that can be called from anywhere. You simply have to use the full name if calling it from outside an actual `QObject`: `QObject::tr(...)` :thinking: 
Thus if you don't need other Qt features (e.g. signal/slot mechanism) I think we can (and should) leave the Registrar as a non QObject type :point_up:

Reply:
Sure. That will be much neater. Will do.